src/make_watertight/MakeWatertight.cpp

```diff
@@ -0,0 +1,1984 @@
+        if (10 * FACET_TOL < move_dist) {
+          std::cout << "  warning6: surf " << surf_id << " vertex move_dist="
+                    << move_dist << std::endl;
+        }
+        rval = zip->t_joint(normal_tag, edge[e_pos - 1], edge[e_pos], skin[s_pos], debug);
+        MB_CHK_SET_ERR(rval, "tjoint failed a");
+        skin.insert(skin.begin() + s_pos, edge[e_pos]);
+        e_pos++;
+        s_pos++;
+      } else { // skin is next: move the t to the curve
+        if (debug) {
+          std::cout << "  zip phase: inserting projected skin vert "
+                    << skin[s_pos] << " between edge verts "
+                    << edge[e_pos - 1] << " and " << edge[e_pos] << std::endl;
+        }
+        double move_dist;
+        rval = gen->line_point_dist(edge[e_pos - 1], edge[e_pos], skin[s_pos], move_dist);
+        MB_CHK_SET_ERR(rval, "could not get line_point_dist");
+        if (10 * FACET_TOL < move_dist) {
+          std::cout << "  warning6: surf " << surf_id << " vertex move_dist="
+                    << move_dist << std::endl;
+        }
+        rval = zip->t_joint(normal_tag, edge[e_pos - 1], skin[s_pos], edge[e_pos], debug);
+        MB_CHK_SET_ERR(rval, "tjoint failed b");
+        edge.insert(edge.begin() + e_pos, skin[s_pos]);
+        e_pos++;
+        s_pos++;
+      }
+    }
+
+
+    // exit if at the end of the curve
+    if (e_pos == edge.size() || s_pos == skin.size())
+      break;
+
+    // The smallest edge length should be no less than MERGE_TOL.
+    if (2 <= e_pos) {
+      double d;
+      rval = gen->squared_dist_between_verts(edge[e_pos - 1], edge[e_pos - 2], d);
+      MB_CHK_SET_ERR(rval, "could not get dist");
+      if (TOL_SQR > d) {
+        std::cout << "zip_loop: d=" << d << std::endl;
+        gen->print_vertex_coords(edge[e_pos - 1]);
+        gen->print_vertex_coords(edge[e_pos - 2]);
+        std::cout << "warning7: surf " << surf_id
+                  << " adjacent edge points are closer than FACET_TOL" << std::endl;
+      }
+    }
+    // The position should be the same. Do not exceed array bounds when checking.
+    if (e_pos != s_pos) {
+      MB_CHK_SET_ERR(moab::MB_FAILURE, "skin and edge positions do not match");
+    }
+    if (edge[e_pos - 1] != skin[s_pos - 1]) {
+      std::cout << "edge[" << e_pos - 1 << "]=" << edge[e_pos - 1]
+                << " skin[" << s_pos - 1 << "]=" << skin[s_pos - 1] << std::endl;
+    }
+    if (edge[e_pos - 1] != skin[s_pos - 1]) {
+      MB_CHK_SET_ERR(moab::MB_FAILURE, "skin and edge vert does not match");
+    }
+  }
+
+  // The skin and curve should be the same size
+  if (edge.size() != skin.size()) {
+    std::cout << "    surf " << surf_id
+              << " sealed skin and curve are not the same size" << std::endl;
+    if (debug) {
+      std::cout << "edge:" << std::endl;
+      gen->print_loop(edge);
+      std::cout << "skin:" << std::endl;
+      gen->print_loop(skin);
+    }
+    //return moab::MB_FAILURE;
```

Comment:
Potential Bug: It looks like this `if` block covers a failure case, but the `return moab::MB_FAILURE` is commented out here.